From 7a815a22155b2c6bf81fb80bc5a877d93799c816 Mon Sep 17 00:00:00 2001
From: dianlujitao <dianlujitao@lineageos.org>
Date: Tue, 1 May 2018 15:37:08 +0800
Subject: [PATCH 08/15] UsbDeviceManager: Use isNormalBoot() where possible

 * Move isNormalBoot() to outer class so it can be picked by all its
   inner classes, don't copy the logic multiple times.

Change-Id: Ie8781bc94a3ef617bf5b3db00498b95788169e8f
---
 .../android/server/usb/UsbDeviceManager.java    | 17 ++++++++---------
 1 file changed, 8 insertions(+), 9 deletions(-)

diff --git a/services/usb/java/com/android/server/usb/UsbDeviceManager.java b/services/usb/java/com/android/server/usb/UsbDeviceManager.java
index 060e0b6ebbb..4f38333f3e4 100644
--- a/services/usb/java/com/android/server/usb/UsbDeviceManager.java
+++ b/services/usb/java/com/android/server/usb/UsbDeviceManager.java
@@ -409,6 +409,11 @@ public class UsbDeviceManager {
         return mContext.getPackageManager().hasSystemFeature(PackageManager.FEATURE_LEANBACK);
     }
 
+    private static boolean isNormalBoot() {
+        String bootMode = SystemProperties.get(BOOT_MODE_PROPERTY, "unknown");
+        return bootMode.equals(NORMAL_BOOT) || bootMode.equals("unknown");
+    }
+
     private final class UsbHandler extends Handler {
 
         // current USB state
@@ -659,11 +664,6 @@ public class UsbDeviceManager {
             Slog.e(TAG, "Unable to set any USB functions!");
         }
 
-        private boolean isNormalBoot() {
-            String bootMode = SystemProperties.get(BOOT_MODE_PROPERTY, "unknown");
-            return bootMode.equals(NORMAL_BOOT) || bootMode.equals("unknown");
-        }
-
         private boolean trySetEnabledFunctions(String functions, boolean forceRestart) {
             if (functions == null || applyAdbFunction(functions)
                     .equals(UsbManager.USB_FUNCTION_NONE)) {
@@ -1380,8 +1380,7 @@ public class UsbDeviceManager {
                 mOemModeMap.get(bootMode);
         // Check to ensure that the oem is not overriding in the normal
         // boot mode
-        if (overridesMap != null && !(bootMode.equals(NORMAL_BOOT) ||
-                bootMode.equals("unknown"))) {
+        if (overridesMap != null && !isNormalBoot()) {
             Pair<String, String> overrideFunctions =
                     overridesMap.get(usbFunctions);
             if (overrideFunctions != null) {
@@ -1418,9 +1417,9 @@ public class UsbDeviceManager {
     }
 
     public static String getPersistProp(boolean functions) {
-        String bootMode = SystemProperties.get(BOOT_MODE_PROPERTY, "unknown");
         String persistProp = USB_PERSISTENT_CONFIG_PROPERTY;
-        if (!(bootMode.equals(NORMAL_BOOT) || bootMode.equals("unknown"))) {
+        if (!isNormalBoot()) {
+            String bootMode = SystemProperties.get(BOOT_MODE_PROPERTY, "unknown");
             if (functions) {
                 persistProp = "persist.sys.usb." + bootMode + ".func";
             } else {
-- 
2.17.0

